adapters/beachfront/beachfront.go

```diff
@@ -0,0 +1,512 @@
+}
+
+// ---------------------------------------------------
+//              Banner
+// ---------------------------------------------------
+type BeachfrontBannerRequest struct {
+	Slots          []BeachfrontSlot `json:"slots"`
+	Domain         string           `json:"domain"`
+	Page           string           `json:"page"`
+	Referrer       string           `json:"referrer"`
+	Search         string           `json:"search"`
+	Secure         int8             `json:"secure"`
+	DeviceOs       string           `json:"deviceOs"`
+	DeviceModel    string           `json:"deviceModel"`
+	IsMobile       int8             `json:"isMobile"`
+	Ua             string           `json:"ua"`
+	Dnt            int8             `json:"dnt"`
+	User           string           `json:"user"`
+	AdapterName    string           `json:"adapterName"`
+	AdapterVersion string           `json:"adapterVersion"`
+	Ip             string           `json:"ip"`
+}
+
+type BeachfrontSlot struct {
+	Slot     string           `json:"slot"`
+	Id       string           `json:"id"`
+	Bidfloor float64          `json:"bidfloor"`
+	Sizes    []BeachfrontSize `json:"sizes"`
+}
+
+type BeachfrontSize struct {
+	W uint64 `json:"w"`
+	H uint64 `json:"h"`
+}
+
+type BeachfrontResponseSlot struct {
+	CrID  string  `json:"crid"`
+	Price float64 `json:"price"`
+	W     uint64  `json:"w"`
+	H     uint64  `json:"h"`
+	Slot  string  `json:"slot"`
+	Adm   string  `json:"adm"`
+}
+
+func NewBeachfrontBannerRequest() BeachfrontBannerRequest {
+	r := BeachfrontBannerRequest{}
+	r.AdapterName = beachfrontAdapterName
+	r.AdapterVersion = beachfrontAdapterVersion
+
+	return r
+}
+
+func NewBeachfrontVideoRequest() BeachfrontVideoRequest {
+	r := BeachfrontVideoRequest{}
+	r.Cur = append(r.Cur, "USD")
+	r.IsPrebid = true
+
+	return r
+}
+
+func getEndpoint(request *openrtb.BidRequest) (uri string) {
+	for i := range request.Imp {
+		if request.Imp[i].Video != nil {
+			// If there are any video imps, we will be running a video auction
+			// and dropping all of the banner actions.
+			return VideoEndpoint
+		}
+	}
+	return BannerEndpoint
+}
+
+func (a *BeachfrontAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	var beachfrontRequests BeachfrontRequests
+	var reqJSON []byte
+	var uri string
+	var errs = make([]error, 0)
+	var err error
+	var imps int
+
+	uri = getEndpoint(request)
+
+	beachfrontRequests, errs, imps = preprocess(request, uri)
+
+	// These are fatal errors -------------
+	if uri == VideoEndpoint {
+		reqJSON, err = json.Marshal(beachfrontRequests.Video)
+		uri = uri + beachfrontRequests.Video.AppId + VideoEndpointSuffix
+	} else {
+		/*
+			We will get here if request contains no Video imps, though it might have
+			Audio or Native imps as well as banner.
+		*/
+		reqJSON, err = json.Marshal(beachfrontRequests.Banner)
+	}
+
+	if imps == 0 {
+		errs = append(errs, errors.New("No valid impressions were found"))
+		return nil, errs
+	}
+
+	if err != nil {
```

Comment:
I don't see you using `err` anymore. So it should always be `nil`, and this never executes.

Can remove it.